File: ACM2/Services/Backend/FuelAgentClient.swift
import Foundation

struct FuelAgentClient {
    private let backend: BackendClient

    init(backend: BackendClient = BackendClient()) {
        self.backend = backend
    }

    func fetchSummary(
        userId: String,
        profile: VehicleProfile,
        trips: [TripResult],
        fuelLogs: [FuelLog]
    ) async throws -> FuelSummary {
        try await backend.post(
            "/fuel/summary",
            body: FuelSummaryRequest(userId: userId, profile: profile, trips: trips, fuelLogs: fuelLogs)
        )
    }
}

File: mobile/src/services/backend/FuelAgentClient.ts
import { FuelSummaryResponse, FuelSummaryRequest } from "./contracts";
import { BackendClient } from "./BackendClient";

export class FuelAgentClient {
  constructor(private readonly backend = new BackendClient()) {}

  getSummary(request: FuelSummaryRequest): Promise<FuelSummaryResponse> {
    return this.backend.post<FuelSummaryResponse>("/fuel/summary", request);
  }
}
